sdk/core/core-client/src/urlHelpers.ts

```diff
@@ -288,11 +291,13 @@ export function appendQueryParams(
   for (const [name, value] of combinedParams) {
     if (typeof value === "string") {
       searchPieces.push(`${name}=${value}`);
-    } else {
+    } else if (Array.isArray(value)) {
       // QUIRK: If we get an array of values, include multiple key/value pairs
       for (const subValue of value) {
         searchPieces.push(`${name}=${subValue}`);
       }
+    } else {
+      searchPieces.push(`${name}=${value}`);
```

Comment:
So this will cause `undefined` to show up as `name=undefined` in the query string, which seems to match how URLSearchParams handles this:

![image](https://user-images.githubusercontent.com/639216/141355054-877f6d27-2061-425e-b896-100ba574abf3.png)

I'm fine with it, but can we test this change on the service that was impacted by the linked issue before merging?

Reply:
If the service did not like this, I think we will need to do:

```suggestion
    } else if (value === undefined) {
      searchPieces.push(`${name}`);
     } else {
      searchPieces.push(`${name}=${value}`);
```

Reply:
The last `else` should be unreachable, right?

Reply:
I think you're right, we could just throw an error there just in case.

Reply:
> So this will cause `undefined` to show up as `name=undefined` in the query string, which seems to match how URLSearchParams handles this:
> 
> ![image](https://user-images.githubusercontent.com/639216/141355054-877f6d27-2061-425e-b896-100ba574abf3.png)
> 
> I'm fine with it, but can we test this change on the service that was impacted by the linked issue before merging?

Yes. I have tested the changes with the actual iothub service that was causing this issue. It works fine. Thanks